command/agent/config.go

```diff
@@ -463,8 +470,9 @@ func DecodeConfig(r io.Reader) (*Config, error) {
 	// Decode
 	var md mapstructure.Metadata
 	msdec, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
-		Metadata: &md,
-		Result:   &result,
+		Metadata:    &md,
+		Result:      &result,
+		ErrorUnused: true,
```

Comment:
We should leave this `false` and remove all the unused fields from Config. Instead use the `Unused` field in `Metadata`. We can scan that for fields that are not (services, checks, service, check) and error / warn on those. This way we have more fine grain control later when we deprecate flags and stuff too.